test/Identity.FunctionalTests/ManagementTests.cs

```diff
@@ -52,5 +56,77 @@ public ManagementTests(ServerFactory serverFactory)
             var email = emails.SentEmails[1];
             await UserStories.ConfirmEmailAsync(email, client);
         }
+
+        [Fact]
+        public async Task CanChangePassword()
+        {
+            // Arrange
+            var principals = new List<ClaimsPrincipal>();
+            var server = ServerFactory.CreateServer(builder =>
+                builder.ConfigureTestServices(s => s.SetupGetUserClaimsPrincipal(user => principals.Add(user), IdentityConstants.ApplicationScheme)));
+
+            var client = ServerFactory.CreateDefaultClient(server);
+            var newClient = ServerFactory.CreateDefaultClient(server);
+
+            var userName = $"{Guid.NewGuid()}@example.com";
+            var password = "!Test.Password1";
+
+            var index = await UserStories.RegisterNewUserAsync(client, userName, password);
+
+            // Act 1
+            var changedPassword = await UserStories.ChangePasswordAsync(index, "!Test.Password1", "!Test.Password2");
+
+            // Assert 1
+            // RefreshSignIn generates a new security stamp claim
+            Assert.NotEqual(GetSecurityStampClaimValue(principals[0]), GetSecurityStampClaimValue(principals[1]));
+
+            // Act 2
+            await UserStories.LoginExistingUserAsync(newClient, userName, "!Test.Password2");
+
+            // Assert 2
+            // Signing in again with a different client uses the same security stamp claim
+            Assert.Equal(GetSecurityStampClaimValue(principals[1]), GetSecurityStampClaimValue(principals[2]));
+        }
+
+        [Fact]
+        public async Task CanChangePasswordWithExternalLogin()
+        {
+            // Arrange
+            var principals = new List<ClaimsPrincipal>();
+            var server = ServerFactory.CreateServer(builder =>
+                builder.ConfigureTestServices(s => s.SetupTestThirdPartyLogin().SetupGetUserClaimsPrincipal(user => principals.Add(user), IdentityConstants.ExternalScheme)));
+
+            var client = ServerFactory.CreateDefaultClient(server);
+            var newClient = ServerFactory.CreateDefaultClient(server);
+
+            var guid = Guid.NewGuid();
+            var userName = $"{guid}";
+            var email = $"{guid}@example.com";
+
+            // Act 1
+            var index = await UserStories.RegisterNewUserWithSocialLoginAsync(client, userName, email);
+            var claim = GetNameIdentifierClaimValue(principals[0]);
+
+            // Assert 1
+            Assert.NotNull(claim);
+
+            // Act 2
+            var changedPassword = await UserStories.ChangePasswordExternalLoginAsync(index, "!Test.Password2");
+            await UserStories.LoginExistingUserAsync(newClient, email, "!Test.Password2");
+
+            // Assert 2
+            // Signing in again with a different client uses the same name identifier claim
+            Assert.Equal(claim, GetNameIdentifierClaimValue(principals[0]));
```

Comment:
What does the name has to be with these changes? Don't you have to be checking the security timestamp?

Reply:
I think this is because the test is tracking the external cookie due to .SetupGetUserClaimsPrincipal(user => principals.Add(user), IdentityConstants.ExternalScheme) isntead of the normal application cookie